packages/store/addon/-private/system/core-store.ts

```diff
@@ -3267,12 +3309,23 @@ abstract class CoreStore extends Service {
     }
 
     // final fallback, no model specific serializer, no application serializer, no
-    // `serializer` property on store: use json-api serializer
+    // `serializer` property on store: use JSON serializer
     serializer = _serializerCache['-default'] || owner.lookup('serializer:-default');
+    if (DEBUG && serializer === undefined) {
+      const JSONSerializer = require('@ember-data/serializer/json').default;
+      owner.register('serializer:-default', JSONSerializer);
+      serializer = owner.lookup('serializer:-default');
+    }
+
     assert(
-      `No serializer was found for '${modelName}' and no 'application', Adapter.defaultSerializer, or '-default' serializer were found as fallbacks.`,
+      `No serializer was found for '${modelName}' and no 'application' serializer was found as a fallback`,
       serializer !== undefined
     );
+    deprecate(`Usage of the '-default' serializer is deprecated.`, false, {
+      id: 'ember-data:default-serializer-fallback',
+      until: '3.12.0',
```

Comment:
the `until` here should be `4.0` but more importantly this should probably be in a different commit since it implements https://github.com/emberjs/rfcs/pull/522 cc @pete-the-pete